application/src/components/ReefForm.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useEffect, useState } from "react";
+import { StorageBackedInput } from "../utils/FolderStorage";
+import ScouterInput from "../scouter/ScouterInput";
+import { InputProps } from "../scouter/ScouterInput";
+
+interface Level {
+  S: number;
+  F: number;
+}
+interface Levels{
+  level1: Level;
+  level2: Level;
+  level3: Level;
+  level4: Level;
+}
+
+
+class ReefScoring extends ScouterInput <Levels, {}, {levels: Levels, undoStack: string[]}> {
+  create(): React.JSX.Element {
+    return <ReefScoring {...this.props} />;
+
+  }
+
+  initialValue(): Levels {
+    return {
+      level1: { S: 0, F: 0 },
+      level2: { S: 0, F: 0 },
+      level3: { S: 0, F: 0 },
+      level4: { S: 0, F: 0 },
+    };
+  }
+  getStartingState(props: InputProps<Levels>): { levels: Levels; undoStack: string[]; } | undefined {
+    return { levels:  {level1: { S: 0, F: 0 },
+    level2: { S: 0, F: 0 },
+    level3: { S: 0, F: 0 },
+    level4: { S: 0, F: 0 },}, undoStack: []}
+    
+  }
+
+  renderInput(): React.ReactNode {
+    
+  
+    //useEffect(() => {this.storage.set(this.state.levels)}, [this.state.levels]);
+   
+  
+    const handleClick = (e: React.FormEvent, level: number, point: string) => {
+      e.preventDefault();
+
+      const updated = {...this.state.levels}
+      updated[`level${level}`][point]+= 1
+    
+      this.setState({levels: updated})
+
+      this.setState({undoStack: [...this.state.undoStack, `level${level}-${point}`]})
```

Comment:
not important but you can merge these two statements like this:
```suggestion
      this.setState({levels: updated, undoStack: [...this.state.undoStack, `level${level}-${point}`]})
```